packages/swing-store/src/transcriptStore.js

```diff
@@ -0,0 +1,443 @@
+      : sqlGetCurrentSpanMetadata;
+    for (const rec of sql.iterate()) {
+      yield spanArtifactName(rec);
+    }
+  }
+
+  const sqlGetSpanEndPos = db.prepare(`
+    SELECT endPos
+    FROM transcriptSpans
+    WHERE vatID = ? AND startPos = ?
+  `);
+  sqlGetSpanEndPos.pluck(true);
+
+  const sqlReadSpanItems = db.prepare(`
+    SELECT item
+    FROM transcriptItems
+    WHERE vatID = ? AND ? <= position AND position < ?
+    ORDER BY position
+  `);
+
+  /**
+   * Read the items in a transcript span
+   *
+   * @param {string} vatID  The vat whose transcript is being read
+   * @param {number} [startPos] A start position identifying the span to be
+   *    read; defaults to the current span, whatever it is
+   *
+   * @returns {Iterable<string>}  An iterator over the items in the indicated span
+   */
+  function readSpan(vatID, startPos) {
+    let endPos;
+    if (startPos === undefined) {
+      ({ startPos, endPos } = getCurrentSpanBounds(vatID));
+    } else {
+      assert.typeof(startPos, 'number');
+      endPos = sqlGetSpanEndPos.get(vatID, startPos);
+      assert.typeof(
+        endPos,
+        'number',
+        `no transcript span for ${vatID} at ${startPos}`,
+      );
+    }
+    insistTranscriptPosition(startPos);
+    startPos <= endPos || Fail`${q(startPos)} <= ${q(endPos)}}`;
+
+    function* reader() {
+      for (const { item } of sqlReadSpanItems.iterate(
+        vatID,
+        startPos,
+        endPos,
+      )) {
+        yield item;
+      }
+    }
+
+    if (startPos === endPos) {
+      return empty();
+    }
+
+    return reader();
+  }
+
+  /**
+   * Read a transcript span and return it as a stream of data suitable for
+   * export to another store.  Items are terminated by newlines.
+   *
+   * @param {string} vatID  The vat whose transcript is being read
+   * @param {number} [startPos] A start position identifying the span to be read
+   *
+   * @returns {AsyncIterable<Uint8Array>}
+   * @yields {Uint8Array}
+   */
+  async function* exportSpan(vatID, startPos) {
+    for (const entry of readSpan(vatID, startPos)) {
+      yield Buffer.from(`${entry}\n`);
+    }
+  }
+
+  const sqlAddItem = db.prepare(`
+    INSERT INTO transcriptItems (vatID, item, position)
+    VALUES (?, ?, ?)
+  `);
+
+  const sqlUpdateSpan = db.prepare(`
+    UPDATE transcriptSpans
+    SET endPos = ?, hash = ?
+    WHERE vatID = ? AND isCurrent = 1
+  `);
+
+  /**
+   * Append an item to the current transcript span for a given vat
+   *
+   * @param {string} vatID  The whose transcript is being added to
+   * @param {string} item  The item to add
+   */
+  const addItem = (vatID, item) => {
+    ensureTxn();
+    const { startPos, endPos, hash } = getCurrentSpanBounds(vatID);
+    sqlAddItem.run(vatID, item, endPos);
+    const newEndPos = endPos + 1;
+    const newHash = computeItemHash(hash, item);
+    sqlUpdateSpan.run(newEndPos, newHash, vatID);
+    const rec = spanRec(vatID, startPos, newEndPos, newHash);
```

Comment:
Missing `true` here.
```suggestion
    const rec = spanRec(vatID, startPos, newEndPos, newHash, true);
```